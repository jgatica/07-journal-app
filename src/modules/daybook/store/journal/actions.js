import journalApi from "@/api/journalApi";

export const loadEntries = async () => {
  const {data} = await journalApi.get('/entries.json')

  for (let id of Object.keys(data)) {
    console.log(id )
  }

}

export const updateEntries = async () => {

}