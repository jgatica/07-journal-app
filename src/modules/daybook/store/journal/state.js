export default () => ({
  isLoading: true,
  entries: [
    {
      id: '1',
      date: new Date().toDateString(),
      text: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation',
      picture: null,
    },
    {
      id: '2',
      date: new Date().toDateString(),
      text: 'Sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation',
      picture: null,
    },
    {
      id: '3',
      date: new Date().toDateString(),
      text: 'Ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation',
      picture: null,
    },
  ],
})