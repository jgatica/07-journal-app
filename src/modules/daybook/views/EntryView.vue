<template>
  <h1>Hola mundo</h1>
</template>

<script>
export default {
  name: "EntryView"
}
</script>

<style scoped lang="scss">

</style>