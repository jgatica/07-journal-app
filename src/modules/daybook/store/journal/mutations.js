// export const myMutation = (state) => {
//
// }

import state from "@/modules/daybook/store/journal/state";
import entry from "@/modules/daybook/components/Entry.vue";

export const setEntries = (state, entries) => {
  state.entries = [...state.entries, ...entries]
  state.isLoading = false
}

export const updateEntry = (state, entry) => {

  const idx = state.entries.map(e => e.id).indexOf(entry.id)
  state.entries[idx] = entry

}

export const updateEntries = () => {

}

export const addEntry = (state, entry) => {
  // state.entries.unshift(entry)

  state.entries = [entry,...state.entries]
}