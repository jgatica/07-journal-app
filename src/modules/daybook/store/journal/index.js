import state from "@/modules/daybook/store/journal/state";
import * as actions from "@/modules/daybook/store/journal/actions";
import * as getters from "@/modules/daybook/store/journal/getters";
import * as mutations from "@/modules/daybook/store/journal/mutations";

const journalModule = {
  namespaced: true,
  actions,
  getters,
  mutations,
  state,
}

export default journalModule