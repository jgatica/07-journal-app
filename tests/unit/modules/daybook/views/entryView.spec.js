import {shallowMount} from "@vue/test-utils"
import {createStore} from "vuex"

import journal from "@/modules/daybook/store/journal"
import journalState from "../../../mock-data/test-journal-state"
import EntryView from "@/modules/daybook/views/EntryView.vue"

const createVuexStore = (initialState) =>
  createStore({
    modules: {
      journal: {
        ...journal,
        state: {...initialState} // ðŸ‘ˆ Sobrescribe al state que esta desectructurado arriba
      },
    }
  }) // Fin createVuexStore

describe('Pruebas en el EntryView', () => {

  const store = createVuexStore(journalState)
  const mockRouter = {
    push: jest.fn(),
  }

  let wrapper

  beforeEach(() => {
    jest.clearAllMocks()

    wrapper = shallowMount(EntryView, {
      props: {
        id: '-Nlmoug_xoyMBm3jZe40'
      },
      global: {
        mocks: {
          $router: mockRouter,
        },
        plugins: [store],
      },
    })

  })

  it('debe sacar al usuario porque el id no existe', () => {
    wrapper = shallowMount(EntryView, {
      props: {
        id: 'Este id no existe en el STORE'
      },
      global: {
        mocks: {
          $router: mockRouter,
        },
        plugins: [store],
      },
    })

    expect(mockRouter.push).toHaveBeenCalledWith({name: 'no-entry'})

  });

});