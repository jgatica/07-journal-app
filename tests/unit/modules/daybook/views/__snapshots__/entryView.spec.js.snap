// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el EntryView debe mostrar entrada correctamente 1`] = `
<div class="entry-title d-flex justify-content-between p-2">
  <div><span class="text-success fs-3 fw-bold">16</span><span class="mx-1 fs-3">Diciembre</span><span class="mx-2 fs-4 fw-light">2023, Sábado</span></div><input type="file" accept="image/jpeg, image/png" style="display: none;">
  <div><button class="btn btn-danger mx-2"> Borrar </button><button class="btn btn-primary"> Subir foto <i class="fa fa-upload"></i></button></div>
</div>
<hr>
<div class="d-flex flex-column px-3 h-75"><textarea placeholder="¿Qué sucedió hoy?"></textarea></div>
<fab-stub icon="fa-save"></fab-stub>
<!--v-if-->
<!--v-if-->
`;
